cmd/shader_compiler/include/shader_descriptor.hpp

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include "shader_compiler.hpp"
+
+#include <nlohmann/json.hpp>
+using json = nlohmann::json;
+
+namespace shader_compiler
+{
+
+enum class ShaderType
+{
+	Vertex,
+	Fragment,
+	Compute,
+	Geometry,
+	TessellationControl,
+	TessellationEvaluation,
+	RayGeneration,
+	Intersection,
+	AnyHit,
+	ClosestHit,
+	Miss,
+	Callable,
+	Task,
+	Mesh,
+	Count
```

Comment:
What shader type does `Count` represent?

Reply:
Count is just the end of the types... A better name would be "Unknown"